Source/Adapters/Blackjack.Adapters/UserAdapter.cs

```diff
@@ -47,32 +54,37 @@ public UserAdapter(IUserFacade userFacade)
     public async Task CreateUser(string username)
     {
         string soundex = CreateSoundex(username);
-        UserBE userBySoundex = await GetUsers(username);
-        if (userBySoundex != null)
+        UserBE userBE = await GetUserBySoundex(username);
+        if (userBE != null)
         {
             throw new WebException(HttpStatusCode.BadRequest, "Username already exists.");
         }
         await _userFacade.CreateUser(username, soundex, balance: 500);
     }
 
-    public async Task<UserBE> GetUsers(string username)
+    public async Task<UserBE> GetUserBySoundex(string username)
     {
         string soundex = CreateSoundex(username);
-        List<UserBE> usersBySoundex = await _userFacade.GetUsers(soundex);
+        List<UserBE> usersBySoundex = await _userFacade.GetUserBySoundex(soundex);
         return usersBySoundex?.SingleOrDefault(user => user.Username.Equals(username, StringComparison.OrdinalIgnoreCase));
     }
 
+    public async Task<UserBE> GetUserByID(int userID)
+    {
+        return await _userFacade.GetUserByID(userID);
+    }
+
     public async Task<UserBE> UpdateUser(UserBE user)
     {
-        UserBE dbUser = await GetUsers(user.Username);
+        UserBE dbUser = await GetUserByID(user.UserID);
         if (dbUser == null)
         {
             throw new WebException(HttpStatusCode.BadRequest, "User does not exist.");
         }
         dbUser.Username = user.Username;
         dbUser.Balance = user.Balance;
-        List<UserBE> updatedUsers = await _userFacade.UpdateUser(dbUser);
-        return updatedUsers.SingleOrDefault(updatedUser => updatedUser.Username.Equals(user.Username, StringComparison.OrdinalIgnoreCase));
+        await _userFacade.UpdateUser(dbUser);
```

Comment:
You were close, but missed the mark on the previous round of feedback.

The facade method should 

1. Take the BE it receives convert it to a DE
2. Send the DE to the db
3. Turn the DE back to a BE
4. Return the BE to the adapter

The adapter isn't responsible for knowing a object has been updated, that's the Facade. So Facade's UpdateUser method is the one-stop-shop for all your update needs. Whatever it returns is the de-facto fully updated object.